tests/routes/test_assets.py

```diff
@@ -519,3 +520,219 @@ def _delete_s3_files(bucket, prefix):
     for obj in response.get("Contents", list()):
         print("Deleting", obj["Key"])
         s3_client.delete_object(Bucket=bucket, Key=obj["Key"])
+
+
+@pytest.mark.asyncio
+async def test_asset_stats(async_client):
+    dataset = "test_asset_stats"
+    version = "v1.0.0"
+
+    pixetl_output_files_prefix = (
+        f"{dataset}/{version}/raster/epsg-4326/90/27008/percent/"
+    )
+    _delete_s3_files(DATA_LAKE_BUCKET, pixetl_output_files_prefix)
+
+    raster_version_payload = {
+        "is_latest": True,
+        "creation_options": {
+            "source_type": "raster",
+            "source_uri": [f"s3://{DATA_LAKE_BUCKET}/test/v1.1.1/raw/tiles.geojson"],
+            "source_driver": "GeoTIFF",
+            "data_type": "uint16",
+            "pixel_meaning": "percent",
+            "grid": "90/27008",
+            "resampling": "nearest",
+            "overwrite": True,
+            "compute_histogram": True,
+            "compute_stats": True,
+            "no_data": 0,
+        },
+    }
+
+    await create_default_asset(
+        dataset,
+        version,
+        version_payload=raster_version_payload,
+        async_client=async_client,
+        execute_batch_jobs=True,
+    )
+
+    resp = await async_client.get(f"/dataset/{dataset}/{version}/assets")
+    asset_id = resp.json()["data"][0]["asset_id"]
+
+    asset_resp = await async_client.get(f"/asset/{asset_id}/stats")
+    version_resp = await async_client.get(f"/dataset/{dataset}/{version}/stats")
+
+    for resp in (asset_resp, version_resp):
+        assert resp.json()["data"]["bands"][0]["min"] == 1.0
+        assert resp.json()["data"]["bands"][0]["max"] == 1.0
+        assert resp.json()["data"]["bands"][0]["mean"] == 1.0
+        assert resp.json()["data"]["bands"][0]["histogram"]["bin_count"] == 256
+        assert resp.json()["data"]["bands"][0]["histogram"]["value_count"][255] == 0
+        # FIXME: I think the following assertion should be true, but the value always
```

Comment:
did you ever figure this out?

Reply:
Nope. Will Ask Thomas on Monday. I think I probably just don't understand how things work.

Reply:
Please see my comments about histograms above

Reply:
Thomas figured it out: I was over-sampling the pixels in the source data.